float32[]         data          # array of data
